### PR TITLE
[ID1000523] implement the picture home page share, save certificate share

### PR DESCRIPTION
**Pull request by Nixiangyu**
--story=1000523 --user=Nixiangyu
https://www.tapd.cn/60210943/prong/stories/view/1160210943001000523
 
Signed-off-by: Nixiangyu nixiangyu@ofworld.com

### CODE REVIEW

File: components/image/image.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	"github.com/csiabb/donation-service/common/rest"
+	"github.com/csiabb/donation-service/components/wx"
+
+	"github.com/golang/freetype"
+	"github.com/golang/freetype/truetype"
+	"golang.org/x/image/math/fixed"
+)
+
+var (
+	logger = log.MustGetLogger("components/image")
```
Comment (hukunge) [inline components/image/image.go]: logger = log.MustGetLogger("image")

File: components/wx/impl.go [hunk 2]
```diff
@@ -12,24 +12,34 @@ import (
...
+	lineColor = map[string]string{"r": "201", "g": "112", "b": "3"}
+)
+
 const (
 	codeAPI        = "/sns/jscode2session"
 	grantType      = "authorization_code"
 	wxAddress      = "https://api.weixin.qq.com"
 	checkFingerURL = "%s/cgi-bin/soter/verify_signature?access_token=%s"
 	accessTokenURL = "%s/cgi-bin/token?grant_type=client_credential&appid=%s&secret=%s"
+	wxacodeURL     = "https://api.weixin.qq.com/wxa/getwxacodeunlimit?access_token=%s"
```
Comment (hukunge) [inline components/wx/impl.go]: https://api.weixin.qq.com这部分在wxAddress里已经存在了

File: components/wx/impl.go [hunk 3]
```diff
@@ -186,3 +196,43 @@ func GetAccessToken(appID string, secret string) (string, error) {
 	}
 	return body.AccessToken, nil
 }
+
+// GetWXACode ...
+func (c *Client) GetWXACode(token string, scene string) (image.Image, error) {
```
Comment (hukunge) [inline components/wx/impl.go]: GetWXQRCode

File: components/wx/impl.go [hunk 4]
```diff
@@ -186,3 +196,43 @@ func GetAccessToken(appID string, secret string) (string, error) {
...
+	wxURL := fmt.Sprintf(wxacodeURL, token)
+	logger.Debugf("Get wx code url: %v", wxURL)
+	var req = &GetWXACodeRequest{
```
Comment (hukunge) [inline components/wx/impl.go]: GetWXQRCodeRequest

File: components/wx/wx.go [hunk 5]
```diff
@@ -20,6 +21,8 @@ type IWXClient interface {
 	DecryptUserInfo(rawData, encryptedData, signature, iv, ssk string) (ui UserInfo, err error)
 	DecryptPhoneNumber(ssk, data, iv string) (phone PhoneNumber, err error)
 	CheckFinger(finger structs.FingerRequest, accessToken string) (*structs.FingerResponse, error)
+	GetWXACode(token string, scene string) (image.Image, error)
```
Comment (hukunge) [inline components/wx/wx.go]: GetWXQRCode

File: context/context.go [hunk 6]
```diff
@@ -8,6 +8,7 @@ package context
 
 import (
 	"fmt"
+	"github.com/csiabb/donation-service/components/image"
```
Comment (hukunge) [inline context/context.go:11]: "github.com/csiabb/donation-service/components/image"位置不合要求

File: context/context.go [hunk 7]
```diff
@@ -65,6 +67,11 @@ func (c *Context) Init() error {
 		return err
 	}
 
+	err = c.initImageBackend()
+	if nil != err {
+
```
Comment (hukunge) [inline context/context.go]: return呐

File: controllers/image/image.go [hunk 8]
```diff
@@ -8,11 +8,10 @@ package image
 
 import (
 	"fmt"
-	"net/http"
-
 	"github.com/csiabb/donation-service/common/rest"
 	"github.com/csiabb/donation-service/common/utils"
 	"github.com/csiabb/donation-service/structs"
+	"net/http"
```
Comment (hukunge) [inline controllers/image/image.go]: format不对

File: controllers/image/image_test.go [hunk 9]
```diff
@@ -57,15 +74,166 @@ func TestRestHandler_Upload(t *testing.T) {
...
+				URL:       "www.baidu.com/aaa.png",
+				Hash:      "aadkaaka",
+				Index:     "aabbcc",
+				Format:    "png",
+				CreatedAt: time.Now(),
+			},
+		},
+	}, nil)
+
+	/*mockBackend.EXPECT().QueryFundsDetail(gomock.Any()).Return(&models.FundsDetail{
```
Comment (hukunge) [inline controllers/image/image_test.go]: 如果没用了，可以删掉

File: components/image/image.go (hunk 1 above)
Comment (Nixiangyu) [reply components/image/image.go]: 修改

File: components/wx/impl.go [hunk 10]
```diff
@@ -12,24 +12,34 @@ import (
...
+	lineColor = map[string]string{"r": "201", "g": "112", "b": "3"}
+)
+
 const (
 	codeAPI        = "/sns/jscode2session"
 	grantType      = "authorization_code"
 	wxAddress      = "https://api.weixin.qq.com"
 	checkFingerURL = "%s/cgi-bin/soter/verify_signature?access_token=%s"
 	accessTokenURL = "%s/cgi-bin/token?grant_type=client_credential&appid=%s&secret=%s"
+	wxacodeURL     = wxAddress + "/wxa/getwxacodeunlimit?access_token=%s"
```
Comment (hukunge) [inline components/wx/impl.go]: 不应该在这个地方+，应该在使用的地方拼接

File: components/wx/impl.go [hunk 11]
```diff
@@ -186,3 +196,43 @@ func GetAccessToken(appID string, secret string) (string, error) {
 	}
 	return body.AccessToken, nil
 }
+
+// GetWXACode ...
+func (c *Client) GetWXQrCode(token string, scene string) (image.Image, error) {
```
Comment (hukunge) [inline components/wx/impl.go:201]: build failed：
comment on exported method Client.GetWXQrCode should be of the form "GetWXQrCode ...

**Pull request closed without merging**